File: services/bookingservice/src/main/java/com/revamp/booking/model/Booking.java
package com.revamp.booking.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Data
@Document(collection = "bookings")
public class Booking {
    @Id
    private String id;

    private String customerId;
    private String customerName;
    private String customerEmail;

    private String vehicleId; // optional if new vehicle
    private VehicleDetails vehicleDetails; // if provided

    private String serviceType; // Service | Modification

    private LocalDate date;

    // For services
    private String timeSlotId;
    private String timeSlotStart; // HH:mm
    private String timeSlotEnd;   // HH:mm

    // For modifications
    private List<String> neededModifications;

    private Integer estimatedTimeHours; // may be null for service
    private Integer estimatedCost; // integer LKR

    private String instructions;

    private String status; // pending, approved, in_progress, completed

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @Data
    public static class VehicleDetails {
        private String make;
        private String model;
        private Integer year;
        private String registrationNumber;
    }
}

File: frontend/src/types/booking.ts
export type ServiceType = "Service" | "Modification";

export interface TimeSlot {
  id: string;
  startTime: string; // ISO time or HH:mm
  endTime: string;   // ISO time or HH:mm
  isAvailable: boolean;
}

export interface AppointmentRequest {
  serviceType: ServiceType;
  date: string; // yyyy-MM-dd
  timeSlotId?: string; // required for Service
  vehicleId?: string; // existing vehicle
  vehicleDetails?: {
    make: string;
    model: string;
    year: string | number;
    registrationNumber: string;
  };
  neededModifications?: string[]; // ids for Modification flow
  estimatedTimeHours?: number;
  estimatedCost?: number;
  instructions?: string;
}

export interface AppointmentResponse {
  id: string;
  status: "pending" | "approved" | "in_progress" | "completed";
}

export interface ModificationItem {
  id: string;
  name: string;
  estimatedHours: number;
  unitPrice?: number;
  description?: string;
}
